test_std_module/CMakeLists.txt

```diff
@@ -6,7 +6,9 @@ add_executable(type_safe_test_std_module test.cpp)
 target_link_libraries(type_safe_test_std_module PUBLIC type_safe)
 target_include_directories(type_safe_test_std_module PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
-# Hopefully MS will provide a better/automatic/IJW way to use the standard library module
+# Hopefully MS will provide a better/automatic/IJW way to use the standard library module, in other words "import std;" should just work
 if( MSVC )
-    target_sources(type_safe_test_std_module PUBLIC "C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/MSVC/14.35.32019/modules/std.ixx")
-endif()
+    # We use a path relative to the compiler location which should be stable and avoid VS upgrade/version/preview install location issues
+    cmake_path(GET CMAKE_CXX_COMPILER PARENT_PATH  CompilerDir)
+    target_sources(type_safe PUBLIC "${CompilerDir}/../../../modules/std.ixx" )
+endif()
```

Comment:
Missing newline.

Reply:
Done.

Out of interest why do you care about a newline at the end of files. I've never seen any issues with one not being there